requests_toolbelt/downloadutils/stream.py

```diff
@@ -12,17 +14,72 @@
     r';\s*(%s|[^\s;=]+)\s*(?:=\s*(%s|[^;]+))?\s*' % (_QUOTED_STRING_RE,
                                                      _QUOTED_STRING_RE)
 )
+_DEFAULT_CHUNKSIZE = 512
 
 
 def _get_filename(content_disposition):
     for match in _OPTION_HEADER_PIECE_RE.finditer(content_disposition):
         k, v = match.groups()
         if k == 'filename':
-            return v
+            # ignore any directory paths in the filename
+            return os.path.split(v)[1]
     return None
 
 
-def stream_response_to_file(response, path=None):
+@contextlib.contextmanager
+def _response_path_to_fd(response, path=None):
+    """
+    Helper context manager to deal with duck-typing a path, which can be:
+    * a string with the path to a file
+    * a file-like object
+
+    :param response: A Response object from requests
+    :type response: requests.models.Response
+    :param path: *(optional)*, Either a string with the path to the location
+        to save the response content, or a file-like object expecting bytes.
+    :type path: :class:`str`, or object with a :meth:`write`
+    :returns: A 2-tuple of a file object; and the name of the file if one can be determined, else None
+    :rtype: tuple(file, str)
+    :raises: :class:`requests_toolbelt.exceptions.StreamingError`
+    """
+    pre_opened = False
+    fd = None
+    filename = None
+    if path:
+        if isinstance(getattr(path, 'write', None), collections.Callable):
+            pre_opened = True
+            fd = path
+            filename = getattr(fd, 'name', None)
+        elif os.path.isdir(path):
```

Comment:
Why not more simply:

```py
elif not os.path.isdir(path):
    # If the user specified a directory, let the 
    filename = path
    if os.path.exists(filename):
        raise
    if os.path.exists(filename):
        raise exc.StreamingError(...)
    fd = open(filename, 'wb')
```

That said, why are we now going to start allowing people to specify the path to a directory now?